### PR TITLE
fix(datadog_agent source): Remove the `status` field footgun for logs.

### PR DESCRIPTION
**Pull request by silversupreme**
Testing and customer experience have revealed that the `status` field is an unnecessary footgun that trips up attempts to process log messages sourced from the Datadog agent.

Some context, friend: when the Datadog agent collects logs from a [Docker](https://cs.github.com/DataDog/datadog-agent/blob/433bc3f47d0bdc891e06611ea331cc1dfce7bdfa/pkg/logs/internal/parsers/dockerstream/docker_stream.go?q=stderr+language%3AGo+path%3A*%2Fdocker*#L110) or a [Kubernetes](https://cs.github.com/DataDog/datadog-agent/blob/433bc3f47d0bdc891e06611ea331cc1dfce7bdfa/pkg/logs/internal/parsers/kubernetes/kubernetes.go?q=stderr+language%3AGo+path%3A*%2Fkubernetes*#L83) container, it will tag the log line with a status based entirely on the stream that it read that log from. `stdout` will always be tagged `INFO`, and `stderr` will always be tagged `ERROR`, no matter the contents of the line itself.

It then appears that the normal Datadog Agent ingest endpoint strips this attribute out. Vector's does not, and some strange behavior in Datadog's backend means that the `status` attribute will override whatever is parsed from the log line, always, which means that users are in for a surprise when they switch to Vector.

We have a small VRL snippet in the Workshop materials that removes this attribute, but I find this unsatisfactory. **The right thing to do is not the default thing to do** in Vector currently, which if nothing else means that users who do not go through our Workshops or somehow miss that snippet will have invisibly wrong behavior until the wrong moment.

It is obvious that Datadog does not consider this field valid, and this change can operate as a bandaid while I work on a diff to strip its processing out from the Agent itself.

### CODE REVIEW

File: src/sources/datadog/agent/logs.rs [hunk 1]
```diff
@@ -105,7 +105,7 @@ pub(crate) fn decode_log_body(
                 Ok(Some((events, _byte_size))) => {
                     for mut event in events {
                         if let Event::Log(ref mut log) = event {
-                            log.try_insert(path!("status"), status.clone());
+                            // log.try_insert(path!("status"), status.clone());
```
Comment (spencergilbert) [inline src/sources/datadog/agent/logs.rs:108]: Can we leave a comment on why this is removed, or a TODO around what we'd end up doing when it's resolved in the agent or logs backend?

**Review by spencergilbert — CHANGES_REQUESTED** [review]
I'm 50/50 between adding a better comment on the now commented out line and fully deleting it

File: src/sources/datadog/agent/logs.rs (hunk 1 above)
Comment (silversupreme) [reply src/sources/datadog/agent/logs.rs:108]: Ah, yeah, I meant to leave a comment but forgot. Thanks for catching that.

Comment (blt) [reply src/sources/datadog/agent/logs.rs:108]: I would generally prefer not to leave commented-out code in the project. If we aren't going to be collecting `status` -- and we need to clear that its removal doesn't require a deprecation cycle -- then we should nix the line entirely.

**Review by tobz — APPROVED** [review]

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**